CRM/Core/Payment.php

```diff
@@ -1313,6 +1313,34 @@ public function doPayment(&$params, $component = 'contribute') {
     return $result;
   }
 
+  /**
+   * Cancel a recurring subscription.
+   *
+   * Payment processor classes should override this rather than implementing cancelSubscription.
+   *
+   * A PaymentProcessorException should be thrown if the update of the contribution_recur
+   * record should not proceed (in many cases this function does nothing
+   * as the payment processor does not need to take any action & this should silently
+   * proceed. Note the form layer will only call this after calling
+   * $processor->supports('cancelRecurring');
+   *
+   * @param \Civi\Payment\PropertyBag $propertyBag
+   *
+   * @return array
+   *
+   * @throws \Civi\Payment\Exception\PaymentProcessorException
+   */
+  public function doCancelRecurring(PropertyBag $propertyBag) {
+    if (method_exists($this, 'cancelSubscription')) {
+      $message = NULL;
+      if ($this->cancelSubscription($message, $propertyBag)) {
```

Comment:
Before this PR cancelSubscription gets the param "subscriptionId", afterwards it does not. This breaks most current cancelSubscription implementations so we can't merge as is :-(

Reply:
@mattwire  so we stopped passing inn

```      
$cancelParams = ['subscriptionId' => $this->_subscriptionDetails->subscription_id];

``

But instead we have 
```
        $propertyBag->setRecurProcessorID($this->_subscriptionDetails->subscription_id);
```

So subscripttonId should still be available - did you test & find it missing?